Ensure that masked instances can be used to initialize np.ma.MaskedArray

mhvk opened the pull request:
And also can be viewed as such.

This is a small bugfix to reduce incompatibility between `Masked` and `np.ma.MaskedArray`; a goal would be to ensure that `Masked` instances can be used well with `matplotlib`. Although this alone does not solve that, but this gives desired behaviour regardless:
```
In [1]: import numpy as np

In [2]: from astropy.utils.masked import Masked

In [3]: m = Masked([np.inf, 1., 2., np.nan, 3.], mask=[True, False, True, False, False])

In [4]: np.ma.MaskedArray(m)
Out[4]: 
masked_array(data=[--, 1.0, --, nan, 3.0],
             mask=[ True, False,  True, False, False],
       fill_value=1e+20)

In [5]: m.view(np.ma.MaskedArray)
Out[5]: 
masked_array(data=[--, 1.0, --, nan, 3.0],
             mask=[ True, False,  True, False, False],
       fill_value=1e+20)


# without this PR:
masked_MaskedNDArray(data=[——, 1.0, ——, nan, 3.0],
                     mask=[ True, False,  True, False, False],
               fill_value=1e+20)
```

### Checklist for package maintainer(s)
<!-- This section is to be filled by package maintainer(s) who will
review this pull request. -->

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

mhvk commented:
@pllim - do you understand the PR change log failure? Should I have put the bugfix changelog entry under `changes/utils` instead? (I had to make the `utils.masked` directory, so that seems not unlikely...)

pllim commented:
If you want `utils.masked` subdir in the change log, you also need to modify this:

https://github.com/astropy/astropy/blob/b740594dcc9f8aacae0c14a5774e09f0732d1aef/pyproject.toml#L22

pllim commented:
The Windows failure is unrelated as it was reported for other PRs too. @astrofrog , do you know what this is about?

https://github.com/astropy/astropy/runs/4231111807?check_suite_focus=true

```
E       assert 1 == 0
E        +  where 1 = <subprocess.Popen object at 0x000002701EE3BBE0>.returncode

...\wcs\tests\extension\test_extension.py:73: AssertionError
```

mhvk commented:
@pllim - thanks! I added those entries since it seems that otherwise we just follow the labels we have, and I guess `utils.masked` is not that different from `io.ascii`. For completeness, I added `utils.iers` as well.

But unfortunately it still fails. Probably easiest if I just put my changelog entry under `utils` - looking back, I see that is what I did before...

pllim commented on pyproject.toml:
```diff
@@ -139,6 +139,14 @@ write_to = "astropy/_version.py"
+
+    [[tool.towncrier.section]]
+        name = "astropy.utils.masked"
+        path = "utils.masked"
```
Hmm... Maybe the bot is checking pyproject.toml from upstream. If you really want `utils.masked`, I can try to commit this part on `main` first and then you can rebase. Or you can move the change log under `utils` as before. Please let me know what you prefer. I apologize for any inconvenience caused.

https://github.com/pllim/actions-towncrier-changelog/blob/55409754c88f9743fcfc5113decbbcb6d98dedfd/check_changelog.py#L31

mhvk replied:
Hmm, I think in analogy with `io.ascii`, etc., we should really have `utils.masked` and `utils.iers`, since there are corresponding labels. So, maybe go for adding those first? My only worry is that perhaps a difference is that plain `utils` exists, while plain `io` does not.

mhvk commented:
Trying to move this forward, I removed the change in changelog labels.

@nstarman - would you have time to have a look at this?

nstarman reviewed (approved):
Looks good! I've also locally tested the code in the comment. Everything works as advertised.
There aren't any doc changes that should show up in RTD, so approving now.
